react_ujs/index.js

```diff
@@ -112,9 +119,9 @@ var ReactRailsUJS = {
         }
 
         if (hydrate && typeof ReactDOM.hydrate === "function") {
-          component = ReactDOM.hydrate(component, node);
+          renderComponent("hydrate", component, node, props);
         } else {
-          component = ReactDOM.render(component, node);
+          renderComponent("render", component, node, props);
         }
```

Comment:
Assignment to `component` is removed here. It appears unused, since the function returns immediately afterwards. In addition, the return value from `ReactDOM.render` has been marked as [deprecated](https://reactjs.org/docs/react-dom.html#render).

> `ReactDOM.render()` currently returns a reference to the root `ReactComponent` instance. However, using this return value is legacy and should be avoided because future versions of React may render components asynchronously in some cases. If you need a reference to the root `ReactComponent` instance, the preferred solution is to attach a [callback ref](https://reactjs.org/docs/more-about-refs.html#the-ref-callback-attribute) to the root element.